app/controllers/setting_controller.rb

```diff
@@ -70,35 +72,36 @@ def create
             next
           end
           logic(available_items)
-          next_customer_time += exponential(60 / $consumer_per_minute, random_generator)
+          next_customer_time += exponential(60.0 / $consumer_per_minute, random_generator)
           sleep([0, next_customer_time - Time.now].max)
         end
       end
-      $list_of_threads.push(thread)
+      if $debug
+        thread.abort_on_exception = true
```

Comment:
Is setting this at the end of init ok? What if exponential() continues to crash?

Reply:
Typically the "main thread" receives the exception from another thread if thread.join or thread.value is called. We don't notice these exceptions with thread.kill.

If the option thread.abort_on_exception is set to true, the whole program is stopped if an unhandled exception in that thread occurs. This also works if the thread raises an exception before the abort_on_exception is set to true.